patroni/consul.py

```diff
@@ -0,0 +1,230 @@
+from __future__ import absolute_import
+import logging
+import os
+import time
+import six
+
+from consul import ConsulException, NotFound, base, std
+from patroni.dcs import AbstractDCS, Cluster, Failover, Leader, Member
+from patroni.exceptions import DCSError
+from patroni.utils import sleep
+from requests.exceptions import RequestException
+
+logger = logging.getLogger(__name__)
+
+
+class ConsulError(DCSError):
+    pass
+
+
+class HTTPClient(std.HTTPClient):
+
+    def __init__(self, *args, **kwargs):
+        super(HTTPClient, self).__init__(*args, **kwargs)
+        self._patch_default_timeout()
+
+    def _patch_default_timeout(self):
+        request_func = getattr(self.session.request, '__func__' if six.PY3 else 'im_func')
+        defaults_attr_name = '__defaults__' if six.PY3 else 'func_defaults'
+        defaults = list(getattr(request_func, defaults_attr_name))
+        code = request_func.__code__ if six.PY3 else request_func.func_code
+        defaults[code.co_varnames[code.co_argcount - len(defaults):code.co_argcount].index('timeout')] = 5
+        setattr(request_func, defaults_attr_name, tuple(defaults))  # monkeypatching
+
+    def get(self, callback, path, params=None, timeout=None):
```

Comment:
The get function from std.HTTPClient does not have a timeout parameter (https://github.com/cablehead/python-consul/blob/master/consul/std.py), so it'll always be None and the `if timeout is None` check is redundant. 

I think you should add comments explaining what you are doing here, how about:

    # Set a default timeout for the request.session.request function, that is used
    # internally  by the methods request.session.get, request.session.post and
    # others. We monkey-patch here to avoid overriding each individual method from
    # request.sesson.*. By default, the timeout is not set. It means that a new
    # session may hang almost indefinitely waiting for the server to respond,
    # which is not what we want in Patroni.

And for the get function:

    # The get function is overridden to handle a special case of it being called
    # with an index and wait parameters. That form indicates that a user needs to
    # wait for the given key to change its value, with a wait timeout supplied. We
    # don't want our monkey-patched timeout to be less than the value of the wait
    # parameter, therefore, we set it to either the value of wait or a default of 5 minutes.